src/pydantic_avro/base.py

```diff
@@ -90,21 +90,12 @@ def get_type(value: dict) -> dict:
                 # If items in array are a object:
                 if "$ref" in items:
                     tn = tn["type"]
-                # If items in array are a logicalType
+                # Necessary to handle things like logical types, list of lists, and list with union
                 if (
                     isinstance(tn, dict)
-                    and isinstance(tn.get("type", {}), dict)
-                    and tn.get("type", {}).get("logicalType") is not None
+                    and isinstance(tn.get("type", None), (dict, list))
```

Comment:
The change to pull the `type` out of the `dict` representing the types of the elements seemed similar to the logic we are using in the case that handles just a raw dictionary:

```
if isinstance(value_type, dict) and len(value_type) == 1:
    value_type = value_type.get("type")
```

The `dict` case is the equivalent for what we had with the previous `if` statements handling `logicalType` and a list of lists.  However, it also handles `list[dict[str,str]]`.  In the case that it is a list, we end up with just a list of the types which is valid to set directly as the items type.